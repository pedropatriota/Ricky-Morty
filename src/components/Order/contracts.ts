export interface IOrder {
	order?: 'Ascending' | 'Descending';
	handleOrder: () => void;
}
