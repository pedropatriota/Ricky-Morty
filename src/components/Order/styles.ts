import styled from 'styled-components';

export const OrderContainer = styled.div`
	display: flex;
	align-items: center;
	gap: 0.5rem;
	cursor: pointer;
	margin-bottom: 0.5rem;
`;
