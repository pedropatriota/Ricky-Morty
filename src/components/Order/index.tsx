import type { IOrder } from './contracts';
import { OrderContainer } from './styles';
import { ArrowDownUp } from 'lucide-react';

const Order = ({ order, handleOrder }: IOrder) => {
	return (
		<OrderContainer onClick={handleOrder} role="button" tabIndex={0}>
			<ArrowDownUp />
			<span>{!order ? 'Add Order' : order}</span>
		</OrderContainer>
	);
};

export default Order;
