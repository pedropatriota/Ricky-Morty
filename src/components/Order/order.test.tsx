import { screen, render, fireEvent, cleanup, waitFor } from '@testing-library/react';
import Order from '.';

afterEach(() => cleanup);

describe('Order', () => {
	const handleOrder = jest.fn();

	it('should render the initial order correctly', () => {
		render(<Order order={undefined} handleOrder={handleOrder} />);

		const button = screen.getByRole('button', {
			name: /add order/i
		});
		const text = screen.getByText(/add order/i);

		expect(button).toBeInTheDocument();
		expect(text).toBeInTheDocument();
	});

	it('should change the order when clicked', async () => {
		const { rerender } = render(
			<Order order={undefined} handleOrder={handleOrder} />
		);

		const button = screen.getByRole('button', {
			name: /add order/i
		});

		fireEvent.click(button);

		expect(handleOrder).toHaveBeenCalledTimes(1);

		rerender(<Order order="Ascending" handleOrder={handleOrder} />);

		expect(screen.getByText('Ascending')).toBeInTheDocument();

		fireEvent.click(button);

		rerender(<Order order="Descending" handleOrder={handleOrder} />);

		expect(screen.getByText('Descending')).toBeInTheDocument();
	});

	it('should match the snapshot', () => {
		const { container } = render(
			<Order order={undefined} handleOrder={handleOrder} />
		);
		expect(container).toMatchSnapshot();
	});
});
