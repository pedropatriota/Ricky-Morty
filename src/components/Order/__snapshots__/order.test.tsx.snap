// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order should match the snapshot 1`] = `
<div>
  <div
    class="sc-aXZVg Ea-DSp"
    role="button"
    tabindex="0"
  >
    <svg
      class="lucide lucide-arrow-down-up"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="m3 16 4 4 4-4"
      />
      <path
        d="M7 20V4"
      />
      <path
        d="m21 8-4-4-4 4"
      />
      <path
        d="M17 4v16"
      />
    </svg>
    <span>
      Add Order
    </span>
  </div>
</div>
`;
